File: Lab 2/Java lab2/Biblioteca.java
public class Biblioteca {
    private Livro[] livros;
    private int quantidadeLivros;

    public Biblioteca(int capacidade) {
        livros = new Livro[capacidade];
        quantidadeLivros = 0;
    }

    public boolean adicionarLivro(Livro livro) {
        if (quantidadeLivros < livros.length) {
            livros[quantidadeLivros] = livro;
            quantidadeLivros++;
            return true;
        }
        return false;
    }

    public Livro buscarLivro(String titulo) {
        for (int i = 0; i < quantidadeLivros; i++) {
            if (livros[i].getTitulo().equalsIgnoreCase(titulo)) {
                return livros[i];
            }
        }
        return null;
    }

    public void mostrarLivros() {
        if (quantidadeLivros == 0) {
            System.out.println("A biblioteca esta vazia.");
        } else {
            System.out.println("Livros na biblioteca:");
            for (int i = 0; i < quantidadeLivros; i++) {
                System.out.println(livros[i]);
            }
        }
    }
}

File: Lab 2/C++ Lab2 (extra) to so treinando/Biblioteca.cpp
#include "Biblioteca.h"
#include <iostream>

bool Biblioteca::adicionarLivro(const Livro &livro) {
    livros.push_back(livro);
    return true;
}

Livro* Biblioteca::buscarLivro(const std::string &titulo) {
    for (auto &livro : livros) {
        if (livro.getTitulo() == titulo) {
            return &livro;
        }
    }
    return nullptr;
}

void Biblioteca::mostrarLivros() const {
    if (livros.empty()) {
        std::cout << "A biblioteca esta vazia." << std::endl;
    } else {
        std::cout << "Livros na biblioteca:" << std::endl;
        for (const auto &livro : livros) {
            std::cout << livro.toString() << std::endl;
        }
    }
}
